src/storage/kv/transaction.rs

```diff
@@ -796,6 +800,55 @@ impl Transaction {
             Err(e) => Err(e),
         }
     }
+
+    /// Scans a range of keys and returns a vector of tuples containing the key, value, timestamp, and deletion status for each key.
+    pub fn scan_all_versions<'b, R>(
+        &self,
+        range: R,
+        limit: Option<usize>,
+    ) -> Result<Vec<ScanVersionResult>>
+    where
+        R: RangeBounds<&'b [u8]>,
+    {
+        // Convert the range to a tuple of bounds of variable keys.
+        let range = convert_range_bounds(range);
+
+        // Initialize an empty vector to store the results.
+        let mut results = Vec::new();
+
+        // Get a range iterator for the specified range.
+        let snap = self.snapshot.as_ref().unwrap().read();
+        let ranger = snap.range_with_versions(range);
+
+        // Iterate over the keys in the range.
+        for (key, value, _, ts) in ranger {
```

Comment:
If `key` is  declared as `mut`, then there's no need for `let mut key = key;` later:
```rust
        for (mut key, value, _, ts) in ranger {
```

Reply:
Thank you, fixed it